gravitee-apim-console-webui/src/management/settings/api-quality-rules-ng/api-quality-rules-ng.component.ts

```diff
@@ -0,0 +1,248 @@
+          disabled: this.isReadonly('api.quality.metrics.categories.weight'),
+        }),
+        labelsWeight: new UntypedFormControl({
+          value: this.settings.apiQualityMetrics.labelsWeight,
+          disabled: this.isReadonly('api.quality.metrics.labels.weight'),
+        }),
+        functionalDocumentationWeight: new UntypedFormControl({
+          value: this.settings.apiQualityMetrics.functionalDocumentationWeight,
+          disabled: this.isReadonly('api.quality.metrics.functional.documentation.weight'),
+        }),
+        technicalDocumentationWeight: new UntypedFormControl({
+          value: this.settings.apiQualityMetrics.technicalDocumentationWeight,
+          disabled: this.isReadonly('api.quality.metrics.technical.documentation.weight'),
+        }),
+        healthcheckWeight: new UntypedFormControl({
+          value: this.settings.apiQualityMetrics.healthcheckWeight,
+          disabled: this.isReadonly('api.quality.metrics.healthcheck.weight'),
+        }),
+      }),
+    });
+
+    this.formInitialValues = this.apiQualityRulesForm.getRawValue();
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next(true);
+    this.unsubscribe$.unsubscribe();
+  }
+
+  isReadonly(property: string): boolean {
+    return PortalSettingsService.isReadonly(this.settings, property);
+  }
+
+  onSubmit() {
+    const updatedSettings = {
+      ...this.settings,
+      ...this.apiQualityRulesForm.getRawValue(),
+    };
+
+    this.portalSettingsService
+      .save(updatedSettings)
+      .pipe(
+        switchMap(() => this.environmentSettingsService.get()),
+        tap(() => this.snackBarService.success('API Quality details successfully updated!')),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => this.ngOnInit());
+  }
+
+  deleteRule(name: string, id: string) {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData>(GioConfirmDialogComponent, {
+        width: '500px',
+        data: {
+          title: 'Delete manual rule',
+          content: `Are you sure you want to delete manual rule <strong>${name}</strong>?`,
+          confirmButton: 'Delete',
+        },
+        role: 'alertdialog',
+        id: 'deleteManualRule',
+      })
+      .afterClosed()
+      .pipe(
+        filter((confirm) => confirm === true),
+        switchMap(() => this.qualityRuleService.delete(id)),
+        tap(() => {
+          this.snackBarService.success(`“${name}” has been deleted”`);
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => this.ngOnInit());
+  }
+
+  addQualityRule() {
+    this.matDialog
+      .open<ApiQualityRulesNgAddDialogComponent, ApiQualityRulesDialogData, ApiQualityRulesDialogData>(
```

Comment:
```suggestion
      .open<ApiQualityRulesNgAddDialogComponent, ApiQualityRulesDialogData, ApiQualityRulesDialogResult>(
```
i's better to keep ApiQualityRulesDialogData & ApiQualityRulesDialogResult 

event if `ApiQualityRulesDialogResult = ApiQualityRulesDialogData`

Reply:
Done